src/shared/default-on-success.js

```diff
@@ -0,0 +1,22 @@
+import { onlineManager } from '@tanstack/react-query'
+
+export function defaultOnSuccess(callback) {
+    if (typeof callback !== 'function') {
+        return handleDefaultOnSuccess
+    }
+    return function composeSuccessHandlers(...args) {
+        handleDefaultOnSuccess.apply(this, args)
+        callback.apply(this, args)
+    }
+}
```

Comment:
We can address this in post-release work (if we decide that it needs to be addressed at all):

This is a weird pattern to me (passing the callback and having it being called after `handleDefaultOnSuccess`) as this is synchronous code. The current implementation hides the fact when what is called in what order, which leads to code that's harder to reason about. I'd prefer to simply call the function after calling `handleDefaultOnSuccess` as that makes core more understandable:

```js
{
  onSuccess: () => {
    // doesn't accept any parameters, so can call it like this
    handleDefaultOnSuccess()

    // if this needs parameters, they can be defined explicitly in the `onSuccess` callback.
    // Otherwise, if we need to call something before `handleDefaultOnSuccess`,
    // we'd have to either handle that inside `handleDefaultOnSuccess`, too.
    // Or we end up with an inconsistent pattern (on callback before `defaultOnSuccess` and one inside)
    callback() 
  },
}
```

I suppose the idea is that, by using `defaultOnSuccess` on success as a convention, it's guaranteed that the default behavior will be executed, but it's just a convention. It could simply be expected to use `handleDefaultOnSuccess` in the `onSuccess` callback.

Reply:
As I said, this is something we can discuss post-release when looking at refactoring the app. I'll create an issue in the post-release epic with my "concerns", stating that I'd like to discuss it, but wouldn't mind keeping things as they are if that's the consensus

Reply:
I don't really understand the pushback on this though, from an API-perspective (`defaultOnSuccess()`), it's just a regular callback? It's perfectly valid to use callbacks in synchronous code. I understand it might be a bit "weird", since it looks like we are just calling the callback after the default-functionality, but there's a reason - we keep the `this`-context intact. The `this`-context can be valuable in `onSuccess`-handlers to get info about the `query/mutation-key` and other info.
There's also less code than having to manually call the `handleDefaultOnSuccess` in every place we have this.